kong/db/dao/certificates.lua

```diff
@@ -0,0 +1,211 @@
+local singletons = require "kong.singletons"
+local cjson      = require "cjson"
+local utils      = require "kong.tools.utils"
+
+-- Get an array of SNI names from either a string (split+sort),
+-- an array(sort) or ngx.null(return {})
+-- Returns an error if the list has duplicates
+-- Returns nil if input is falsy.
+local function parse_name_list(input, errors)
+  local name_list
+  if type(input) == "string" then
+    name_list = utils.split(input, ",")
+
+  elseif type(input) == "table" then
+    name_list = utils.shallow_copy(input)
+
+  elseif input == ngx.null then
+    name_list = {}
+  end
+
+  if not name_list then
+    return nil
+  end
+
+  local found = {}
+  for _, name in ipairs(name_list) do
+    if found[name] then
+      local msg   = "duplicate sni in request: " .. name
+      local err_t = errors:invalid_input(msg)
+      return nil, tostring(err_t), err_t
+    end
+    found[name] = true
+  end
+
+  table.sort(name_list)
```

Comment:
So we wish to sort the snis alphabetically?

Reply:
The reason I sort them out is that when I added ids to Cassandra the SNIs started being returned in random order (because the random uuid generation). This made writing the tests very painful, and I thought it would confuse users as well.